### PR TITLE
Singularities created with Bags of Holding have a limited lifetime

### PR DESCRIPTION
**Pull request by Tacolizard**
:cl: Tacolizard Forever
tweak: Singularities created by bluespace interference now have a limited lifetime.
add: Coders can now create subtypes of singularity that self-destruct after a time.
/:cl:

Idea from this forum post, read here for more reasoning: https://tgstation13.org/phpBB/viewtopic.php?f=9&t=30&start=3150#p375882

Singularities created by BoH last for 20 seconds before collapsing. the BoH is emagged, they last a minute.

As for the exact length of the sing lifetime, I am open to debate. 

**If the BoH sing eats the SM it will delete its lifetime timer, making its lifetime uncapped.**
this is to discourage BoH bombing a delaminating SM, as the resulting singularity will do more damage than the SM could by just exploding.

### CODE REVIEW

**Comment by Qustinnus** [comment]
Maybe add emag capability to bags of holding for bonus time?

**Comment by Tacolizard** [comment]
@Qustinnus sure. I'll lower the normal lifetime to 20 seconds though

**Comment by PKPenguin321** [comment]
RIP singulos ever fucking happening

**Comment by Tacolizard** [comment]
all bugs fixed, pr is ready for review.
@PKPenguin321 20 seconds is long enough to destroy an entire department, similar to the effect of a syndie bomb. a minute with the emagged version is enough to destroy most of the station, provided the sing doesnt fly into space.

**Comment by Slignerd** [comment]
Unless you're scanned in an autocloner, you're giving up your round to use this. Even if you do have an autocloner, you're giving up most, if not all, of your equipment.

Making it last only 20-60 seconds sounds lame.

**Comment by praisenarsie** [comment]
then don't use it lol

**Comment by Tacolizard** [comment]
if you want to make a sing, sabotage the sm or buy a generator from cargo. 

BoH sings are too low-effort

**Comment by Slignerd** [comment]
Getting a singularity from the supermatter is nigh-impossible as a non-engineer, and extremely unlikely as an engineer.

Assuming you *are* an engineer, even if you disable the scrubbers and unwrench the pipes, even if you somehow get all the cannisters you need to over-pressurize it, and even if you manage not to be spotted in the process, the supermatter still takes a long time to delaminate, and will loudly tip everyone off on the engineering and the common channels.

It's pretty much the same with cargo. Unless it's some sort of open project that all the crew wishes to go along with, then it simply won't happen. Because it means lots of time spent farming points, then ordering what you want inconspicuously, and then lots of setup.

There's a reason both of these barely happen.

**Comment by Qustinnus** [comment]
git gud slig

**Comment by Tacolizard** [comment]
@Slignerd any traitor who wants to actually destroy the station with a singularity will buy an emag and just emag the BoH. The full minute an emag grants is enough to destroy a large majority of the station, and I think many antags will find it worthwhile

File: code/game/objects/items/storage/backpack.dm [hunk 1]
```diff
@@ -82,7 +90,12 @@
 		investigate_log("has become a singularity. Caused by [user.key]", INVESTIGATE_SINGULO)
 		to_chat(user, "<span class='danger'>The Bluespace interfaces of the two devices catastrophically malfunction!</span>")
 		qdel(W)
-		var/obj/singularity/singulo = new /obj/singularity (get_turf(src))
+
+		if(emagged)
+			singulo = new /obj/singularity/boh/emagged (get_turf(src))
```
Comment (MrDoomBringer) [inline code/game/objects/items/storage/backpack.dm]: Why subtype? why not pass the lifespan directly into a new parameter for the boh? less code needed
so:
- singulo = new /obj/singularity/boh/(get_turf(src), 200)

then in sunglarity.dm
- singularity/Initialize(var/turf, var/span = 0)
  - lifespan = span

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/backpack.dm (hunk 1 above)
Comment (Tacolizard) [reply code/game/objects/items/storage/backpack.dm]: well i figure that lifespan might be useful for some of the existing singularity subtypes, or new ones.

**Comment by Kromgar** [comment]
What if the boh singulo eats the supermatter? It should definitely be self sustaining at that point

File: code/game/objects/items/storage/backpack.dm [hunk 2]
```diff
@@ -42,8 +42,14 @@
...
 	var/pshoom = 'sound/items/pshoom.ogg'
 	var/alt_sound = 'sound/items/pshoom_2.ogg'
+	var/emagged = FALSE
+	var/obj/singularity/boh/singulo
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 60, "acid" = 50)
 
+/obj/item/storage/backpack/holding/examine(mob/user)
+	..()
+	if(emagged)
+		to_chat(user, "A black scorch mark covers the bluespace transceiver")
```
Comment (PKPenguin321) [inline code/game/objects/items/storage/backpack.dm]: .

**Comment by PKPenguin321** [comment]
if you MUST do this at least make it start at a stage higher than 2

**Comment by Tacolizard** [comment]
@PKPenguin321 alright

**Comment by Dorsisdwarf** [comment]
Does this mean that bags of holding lose the stupid gamey safety popup bullshit now that they're only 20 seconds of life?

**Comment by Tacolizard** [comment]
@Dorsisdwarf no

I could do that if you want

**Comment by Slignerd** [comment]
Will you implement Kromgar's suggestion of the singularity persisting if it eats a supermatter crystal?

**Comment by Tacolizard** [comment]
@Slignerd I'm considering it. If more people like that suggestion I will implement it.

**Comment by PKPenguin321** [comment]
It would be good if only because then you couldn't just pop a BoH bomb on a SM that's about to explode and walk away relatively penalty free

**Comment by Tacolizard** [comment]
@PKPenguin321 it wouldn't be penalty free. You'd get gibbed as soon as you did it and most of engineering would be destroyed, and that's without an emag

**Comment by Kromgar** [comment]
I think a 20s singulo would do less damage than a supermatter explosion. Atmos might still be intact

**Comment by Tacolizard** [comment]
@Kromgar you might be forgetting that it spawns in at stage 3.

although, yeah, because there is randomness in how the sing moves, atmos might still be intact.

**Comment by PKPenguin321** [comment]
>not scanning in a cloner before BoH bombing

**Comment by Tacolizard** [comment]
@PKPenguin321 even so,

thats a lotta damage

**Comment by PKPenguin321** [comment]
more like "thats not as much damage as the supermatter would have done"

**Comment by Tacolizard** [comment]
@PKPenguin321 alright. now, if the BoH sing eats the SM it will delete its lifetime timer, making its lifetime uncapped.

**Comment by YakumoChen** [comment]
how do you emag a boh anyway?

**Comment by Wilchenx** [comment]
I do not like this change, bohbombing is mainly used against antags as desperate move, limiting the time singularity is alive means you don't even need to kill it with other bohbombs, it just disappears.

The main argument of the forum post was short rounds due to bohbombs. This was already adressed by nerfing bluespace technodes making it a lot more costly. There is no need to nerf bohbombs to the fucking ground.

**Comment by Tacolizard** [comment]
@Wilchenx I'm pretty sure bohbombing as a nonantag is bannable? Isn't it similar to bombing an antag as a nonantag?

**Comment by Wilchenx** [comment]
Not if you kill a antag like wizard or 1+ nukeops

**Comment by Durkel** [comment]
This is a good change and should have been done ages ago. There's no reason a round should be brought to a screeching halt because someone managed to get some diamonds, bluespace crystals,gold and waited 15 minutes for RD points.

Twenty seconds of singulo is still enough to completely fuck over a shuttle which is where most singulo bombs happens anyway.

**Comment by Tacolizard** [comment]
ITT: people who underestimate twenty seconds of an instant stage-3 singularity

**Comment by CPTANT** [comment]
There isn't really a reason why 2 BoH's should create a singularity in the first place right?

They could just as well create a huge explosion or have some fancy unique effect.

**Comment by Qustinnus** [comment]
https://roll20.net/compendium/dnd5e/Bag%20of%20Holding

**Comment by CPTANT** [comment]
> Some items should not be put in a bag of holding as they may make the bag explode, destroying all its contents. These items are:
> 
>     other bags of holding
>     bags of tricks with charges remaining
>     wands of cancellation with charges remaining
>     other bags containing any of the above items (these have a chance of not exploding if you nest them; see below)

https://nethackwiki.com/wiki/Bag_of_holding

**Comment by Qustinnus** [comment]
A singulo gets closer to a rift than an explosion.

**Comment by MimicFaux** [comment]
How does one emag a BoH, and what if you just want to store the emag, and not use it

**Comment by ExcessiveUseOfCobblestone** [comment]
PLEASE DO NOT GIVE THE HANDWAVE TRAITOR ITEM MORE FUNCTIONALITY

**Comment by Tacolizard** [comment]
@ExcessiveUseOfCobblestone I'm not in favor of it, but I'm not sure how many people do want the emag function. I'll change it if it means more people like this pr

**Comment by pigeonsk** [comment]
The singularity, as well its ensuing death and destruction, are iconic to SS13, and I'd honestly be extremely sad to see it go like this. I've long been a critic of the SM engine, and others here have explained why SM singulos are a nonstarter pretty well. I agree that BoH bombing is low effort and shouldn't merit the death and destruction that it does, whereas sabotaging a resiliently proofed singulo setup could be difficult. I'd prefer this PR to be merged and the singularity engine to be brought back but that ship has long sailed.

RIP Lord Singuloth indeed.

**Comment by pigeonsk** [comment]
Sorry I'm just emotionally attached to the singularity and after a bit of thinking I realize that fact, in and of itself, is not a good reason to reject a needed balance PR of low-effort station annihilation.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Tacolizard you can have the emag function, just not on the emag ;)

**Comment by KorPhaeron** [comment]
@pigeonsk I think the singularity is one of the defining classics of ss13, and this will make it almost non existent combined with our engine changes. Perfectly valid to feel strongly about that, balance should be secondary to fun.

Also yeah @ExcessiveUseOfCobblestone is right, if nothing else no more nonsense emag functions

**Comment by Cyberboss** [comment]
I'd prefer to see an ingame way of controlling a singloose rather than this

**Comment by AlexTheSysop42** [comment]
Maybe make BoHs deal much more damage to singulos, so anyone brave enough can destroy it

**Comment by KorPhaeron** [comment]
Yeah, it used to be that a single bag of holding would wipe out a singularity if thrown into it. Making singularities take significantly more damage from bombs instead of regenerating instantly might be nice.

**Comment by GuyonBroadway** [comment]
Killing the sing is killing a part of SS13 heritage, the BoH bomb was always a risky venture and time was it took less effort than a roundstart singuloose. 

The only time I ever see BoH bombs is when one lucky enough scientist gets to the summon nar-sie rune in the nick of time to shut down the cult, and even then its hugely fucking obvious when a science nerd is running to the magical round end crescendo device with a BoH in hand what he is going to do. 

It's sad enough that the sing was removed as an option for engineering (and worse still that it was nerfed into near uselessness due to the radiation changes) there was something very ss13 about having a volatile ball of "U dun fukked" powering the nonsensical metal death trap masquerading as a space station.

It feels like people are starting more and more to forget the spirit on which ss13 was founded in favour of nerfs and hugboxing and coders doing overly complex "fixes" for things just because they felt they had the skill to do so without thinking about if the fix was even necessary in the first place.


Now really think for a second, when was the last time that you had a round ruined by a BoH bomb? When was the last time a BoH sing was really REALLY an issue? It could be that on hippie people have a different kind of autism but we were basically "TG EU but people say "faggot" instead of hello", MSO even occasionally called hippie "TG's third server".

I really don't think hugboxing the game is the way forward, offer counterplay yes, offer a chance to prevent yes but not something that self limits because that limits the scope of what the player can do.

**Comment by KorPhaeron** [comment]
I'd accept a PR that made singularities easier to kill, but having them fizzle out on their own is dull

**Pull request closed without merging**